mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -229,6 +229,26 @@ static Optional<Type> convertVecType(sycl::VecType type,
   return convertBodyType("class.sycl::_V1::vec", type.getBody(), converter);
 }
 
+/// Converts SYCL atomic type to LLVM type.
+static Optional<Type> convertAtomicType(sycl::AtomicType type,
+                                        LLVMTypeConverter &converter) {
+  auto convertedTy = LLVM::LLVMStructType::getIdentified(
+      &converter.getContext(), "class.sycl::_V1::atomic");
+  auto elementType = LLVM::LLVMPointerType::get(type.getDataType(),
+                                                (int)(type.getAddrSpace()));
+
+  if (!convertedTy.isInitialized()) {
+    if (failed(convertedTy.setBody(elementType, /*isPacked=*/false)))
+      return llvm::None;
+  } else if (elementType != convertedTy.getBody()[0]) {
+    // If the name is already in use, create a new type.
+    convertedTy = LLVM::LLVMStructType::getNewIdentified(
+        &converter.getContext(), "class.sycl::_V1::atomic", elementType,
+        /*isPacked=*/false);
+  }
+  return convertedTy;
```

Comment:
```suggestion
return convertBodyType("class.sycl::_V1::atomic", type.getBody(), converter);
```

More in line with other types. (Also save some lines)